src/RequestHandler.ts

```diff
@@ -97,34 +98,34 @@ async function handleFetchAsPromise<T extends HttpMethodTemplate>(
   const url = getUrl(opts);
   log(`_${method.toLowerCase()}`, url);
   const reqOpts = requestConfigToFetchConfig(method, opts);
-  const context = { method, url };
-  const errorContext = { errorToThrow: opts.errorWithCleanerStack, context: { method, url } };
+  const errorContext: ErrorContext = { method, url };
+
   let response: Response;
   try {
     response = await fetch(url, reqOpts);
   } catch (e: any) {
     // This error only occurs when there are network errors and such
-    throw opts.errorWithCleanerStack.addContext(context).setCause(e);
+    throw opts.errorWithCleanerStack.addContext(errorContext).setCause(e);
   }
-  errorContext.errorToThrow.statusCode = response.status;
+  opts.errorWithCleanerStack.statusCode = response.status;
```

Comment:
we should probably leave a comment here as to why we haven't included it in the `errorContext` (ie. to not break TDBJS)